### PR TITLE
substitute casts with appropriate functions

### PR DESCRIPTION
**Pull request by kirbysayshi**

### CODE REVIEW

File: source/main.c [hunk 1]
```diff
@@ -182,10 +182,10 @@ int test2() {
...
     double t = 1.0 / sampling_rate;
     for (int i = 0; i < num_frames; i++) {
-      float sig1 = (float)(v1 * sin(TWOPI * f1 * i * t));
-      float sig2 = (float)(v2 * sin(TWOPI * f2 * i * t));
-      float sig3 = (float)(v3 * sin(TWOPI * f3 * i * t));
-      float sig4 = (float)(v4 * sin(TWOPI * f4 * i * t));
+      double sig1 = v1 * sin(TWOPI * f1 * i * t);
+      double sig2 = v2 * sin(TWOPI * f2 * i * t);
+      double sig3 = v3 * sin(TWOPI * f3 * i * t);
+      double sig4 = v4 * sin(TWOPI * f4 * i * t);
```
[the remark this answers is not in the document]
Comment (drubinstein) [reply source/main.c:188]: Unfortunately, I think the double is important here. 
Float becomes: 2.2675737E-5
Double become: 2.2675736961451248E-5

When it comes to sample timing I find that precision is kind of important. You can convert to float but I recommend you keep the argument to `sin` as double. 

[To try it yourself](http://weitz.de/ieee/)

[the remark this answers is not in the document]
Comment (kirbysayshi) [reply source/main.c:188]: Ok. Just to make sure I understand your feedback, you think this PR as-is is valid (using double)?

[the remark this answers is not in the document]
Comment (drubinstein) [reply source/main.c:188]: yea.

[the remark this answers is not in the document]
Comment (kirbysayshi) [reply source/main.c:188]: Nice! Thanks for the feedback!

**Pull request merged**